doc/introduction/interfaces/numpy.rst

```diff
@@ -125,21 +125,19 @@ How does PennyLane know which arguments of a quantum function to differentiate,
 For example, you may want to pass arguments to a QNode but *not* have
 PennyLane consider them when computing gradients.
 
-All positional arguments provided to the QNode are assumed to be differentiable
-by default, but this behaviour is deprecated. In a future release, only arguments explicitly
-marked as trainable or marked using the `argnum` keyword argument will be treated as trainable. Not explicitly marking arguments currently raises a 
-deprecation warning.
-
-For now, arguments are internally turned into arrays from the PennyLane NumPy module,
-which have a special flag ``requires_grad`` specifying whether they are trainable or not:
+Regular positional arguments provided to the QNode are not assumed to be differentiable
+by default. This includes arguments in the form of built-in Python data types, and arrays from
+the original NumPy module. Thus, arguments need to be explicitly marked as trainable or selected
+using the `argnum` keyword. To mark an argument as trainable, a special flag ``requires_grad``
```

Comment:
```suggestion
using the ``argnum`` keyword. To mark an argument as trainable, a special flag ``requires_grad``
```

Reply:
Quick question, is the `argnum` fix in the `qml.grad` and `qml.jacobian` function in the stil open PR required for this to work?

That is, if you have `qml.grad(fn, argnum=0)(x, y)` where `x = np.array(0.5, requires_grad=False)`?

Reply:
Actually nevermind, I thought about it some more, and I don't think this is needed yet :)